lib/AST/ASTStructuralEquivalence.cpp

```diff
@@ -171,18 +176,18 @@ static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
     return true;
 
   case TemplateArgument::Type:
-    return Context.IsStructurallyEquivalent(Arg1.getAsType(), Arg2.getAsType());
```

Comment:
This is one typical example when we falsely called the toplevel interface function.